oodeel/extractor/torch_feature_extractor.py

```diff
@@ -237,9 +250,28 @@ def predict(
                     f if features[i] is None else torch.cat([features[i], f], dim=0)
                 )
 
+            # Get labels
+            if return_labels and contains_labels:
+                lbl_batch = elem[1]  # labels must be at index 1 in the batch tuple
+                # If labels are one-hot encoded, take the argmax
+                if len(lbl_batch.shape) > 1 and lbl_batch.shape[1] > 1:
+                    lbl_batch = lbl_batch.view(lbl_batch.size(0), -1)
+                    lbl_batch = torch.argmax(lbl_batch, dim=1)
+                # If labels are in two dimensions, squeeze them
+                if len(lbl_batch.shape) > 1:
+                    lbl_batch = lbl_batch.view([lbl_batch.shape[0]])
```

Comment:
Maybe make a function out of it and use it twice, here and in the case of  `isinstance(dataset, get_args(ItemType)) == True`